### PR TITLE
Gra 945 gtm bridge

### PR DESCRIPTION
**Pull request by khmMouna**

### CODE REVIEW

File: GoogleTagManagerBridge/GoogleTagManagerBridge/AppDelegate.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+//
+//  AppDelegate.h
+//  GoogleTagManagerBridge
+//
+//  Created by Bastien MATTHAI on 18/03/2019.
```
Comment (oristanovic) [inline GoogleTagManagerBridge/GoogleTagManagerBridge/AppDelegate.h]: to remove

File: GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.h [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ */
+
+
+#import <Foundation/Foundation.h>
+#import "Firebase/Firebase.h"
+#import "GoogleTagManager/GoogleTagManager.h"
+
+
+/* Accengage Actions */
+FOUNDATION_EXPORT NSString *const ACTION_TRACK_EVENT;
```
Comment (oristanovic) [inline GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.h]: there's a warning on this line

File: GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+    
+    NSString *acccengageAction = parameters[KEY_ACTION];
+    if (!acccengageAction) {
+        NSLog(@"ACCCustomTagProvider | No Accengage action found in parameters");
+        return nil;
+    }
+    
+    NSMutableDictionary *mutableDictionnary = [[NSMutableDictionary alloc] initWithDictionary:parameters];
+    [mutableDictionnary removeObjectForKey:KEY_ACTION];
+    NSDictionary *accengageParamters = mutableDictionnary;
```
Comment (oristanovic) [inline GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m]: mutableDictionary.copy

File: GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+///--------------------------------------
+
+- (NSObject*)executeWithParameters:(NSDictionary*)parameters {
+    
+    if (!parameters) {
+        NSLog(@"ACCCustomTagProvider | No parameters found");
+        return nil;
+    }
+    
+    NSString *acccengageAction = parameters[KEY_ACTION];
```
Comment (oristanovic) [inline GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m]: accengage
Also, check typos like this in all classes...

File: GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+    
+    NSMutableDictionary *mutableDictionnary = [[NSMutableDictionary alloc] initWithDictionary:parameters];
+    [mutableDictionnary removeObjectForKey:KEY_ACTION];
+    NSDictionary *accengageParamters = mutableDictionnary;
+    
+    if ([acccengageAction isEqualToString:ACTION_TRACK_EVENT]) {
+        
+        [self parseTrackEventMessage:accengageParamters];
+        
+    } else if([acccengageAction isEqualToString:ACTION_TRACK_LEAD]) {
```
Comment (oristanovic) [inline GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m]: style (spacing)
also, check all other methods

File: GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m [hunk 6]
```diff
@@ -0,0 +1,454 @@
...
+        if (error) {
+            NSLog(@"ACCCustomTagProvider | Delete UDI error %@", error.localizedDescription);
+        }
+    }];
+    
+}
+
+- (void)parseTrackIncrementUdi:(NSDictionary *)parameters {
+    
+    ACCDeviceInformationSet * deviceInformationSet = [[ACCDeviceInformationSet alloc] init];
```
Comment (oristanovic) [inline GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m]: *deviceInformationSet

File: GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m [hunk 7]
```diff
@@ -0,0 +1,454 @@
...
+    NSMutableArray *items = [[NSMutableArray alloc] init];
+    
+    [jsonArray enumerateObjectsUsingBlock:^(id  _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
+        
+        if (![obj isKindOfClass:[NSDictionary class]]) {
+            NSLog(@"ACCCustomTagProvider | Object is not a dictionnary");
+            return;
+        }
+
+        NSDictionary *item = obj;
```
Comment (oristanovic) [inline GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m]: dictionaryWithDictionary seems nicer here

File: GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m [hunk 8]
```diff
@@ -0,0 +1,454 @@
...
+    
+    [cartItems enumerateObjectsUsingBlock:^(id  _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
+        
+        if (![obj isKindOfClass:[ACCCartItem class]]) {
+            NSLog(@"ACCCustomTagProvider | The object is not an ACCCartItem objet");
+            return;
+        }
+        
+        ACCCartItem *cartItem = obj;
+        NSMutableDictionary *item = [[NSMutableDictionary alloc] init];
```
Comment (oristanovic) [inline GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m]: Use literals where possible 
NSMutableDictionary *item = @{}.mutableCopy

File: GoogleTagManagerBridge/GoogleTagManagerBridge/AppDelegate.h (hunk 1 above)
Comment (khmMouna) [reply GoogleTagManagerBridge/GoogleTagManagerBridge/AppDelegate.h]: Done

File: GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.h (hunk 2 above)
Comment (khmMouna) [reply GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.h]: Done

File: GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m (hunk 3 above)
Comment (khmMouna) [reply GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m]: Done

File: GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m (hunk 4 above)
Comment (khmMouna) [reply GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m]: Done

File: GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m (hunk 5 above)
Comment (khmMouna) [reply GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m]: Done

File: GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m (hunk 6 above)
Comment (khmMouna) [reply GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m]: Done

File: GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m (hunk 7 above)
Comment (khmMouna) [reply GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m]: Done

File: GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m (hunk 8 above)
Comment (khmMouna) [reply GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m]: Done

**Comment by oristanovic** [comment]
Can you please re-check the non resolved comments ?

**Pull request merged**